From 7bee934617fbfdeca21575a870c693056aef3e41 Mon Sep 17 00:00:00 2001
From: Nalin Dahyabhai <nalin@redhat.com>
Date: Thu, 1 Oct 2015 18:59:34 -0400
Subject: [PATCH] Set plugin_base_dir for kadmin tests

In the krb5.conf used by the kadmin tests, include a plugin_base_dir
setting.  Otherwise the KDC can load and run code from kdcpreauth
modules in the install tree.

[ghudson@mit.edu: commit message]

(cherry picked from commit 3db59d81bbbe389553d13efeee1c99117b459b24)

ticket: 8262
version_fixed: 1.14
status: resolved

(cherry picked from commit 7861885ffa0dcccfbb57b7ad7a0ccff445380253)
[rharwood@redhat.com: this didn't quite right]
---
 src/kadmin/testing/proto/krb5.conf.proto | 1 +
 src/kadmin/testing/scripts/start_servers | 1 +
 2 files changed, 2 insertions(+)

diff --git a/src/kadmin/testing/proto/krb5.conf.proto b/src/kadmin/testing/proto/krb5.conf.proto
index c2648d6..e9702bb 100644
--- a/src/kadmin/testing/proto/krb5.conf.proto
+++ b/src/kadmin/testing/proto/krb5.conf.proto
@@ -2,6 +2,7 @@
 	default_realm = __REALM__
 	default_keytab_name = FILE:__K5ROOT__/v5srvtab
 	dns_fallback = no
+	plugin_base_dir = __PLUGIN_DIR__
 
 [realms]
 	__REALM__ = {
diff --git a/src/kadmin/testing/scripts/start_servers b/src/kadmin/testing/scripts/start_servers
index 80cf8d6..52e38dd 100755
--- a/src/kadmin/testing/scripts/start_servers
+++ b/src/kadmin/testing/scripts/start_servers
@@ -40,6 +40,7 @@ if [ $local = 0 ]; then
 		-e "s/__KDCHOST__/$hostname/g" \
 		-e "s/__LOCALHOST__/$localname/g" \
 		-e "s#__MODDIR__#$TOP/../util/fakedest$libdir/db-modules#g"\
+                -e "s#__PLUGIN_DIR__#$TOP/../util/fakedest$libdir/plugins#g"\
 		< $STESTDIR/proto/krb5.conf.proto > $K5ROOT/krb5.conf
 
 # Using /usr/ucb/rsh and getting rid of "-k $REALM" until we get
-- 
2.9.3

